src/extensionProvider.ts

```diff
@@ -93,65 +93,39 @@ export class ExtensionProvider implements IDappProvider {
     return !!this.account;
   }
 
-  async sendTransaction(transaction: ITransaction): Promise<ISignedTransaction> {
-    const txResponse = await this.startBgrMsgChannel("sendTransactions", {
-      from: this.account.address,
-      transactions: [transaction.toPlainObject()],
-    });
-
-    return this.getTransactionFactory().fromPlainObject(txResponse[0]);
+  async signTransaction<T extends ITransaction>(transaction: T): Promise<T> {
+    let response = await this.signTransactions([transaction]);
```

Comment:
maybe we should check for nulls here, to not throw an unhandled, we could throw a custom message if the user forgot to include the transaction

Reply:
The input parameter is not optional. In case of TypeScript code, the user should not be able to compile without passing a transaction. 

But you are right about adding improving the error handling / throwing, overall. Is it OK to postpone this to a separate PR?